src/components/contextual/pages/vebal/MyVebalInfo.vue

```diff
@@ -0,0 +1,384 @@
+<script setup lang="ts">
+import { useLock } from '@/composables/useLock';
+import { bnum } from '@/lib/utils';
+import useWeb3 from '@/services/web3/useWeb3';
+import useNumbers from '@/composables/useNumbers';
+import { differenceInDays, format } from 'date-fns';
+import { PRETTY_DATE_FORMAT } from '@/components/forms/lock_actions/constants';
+import rank from '@/assets/images/icons/rank.svg';
+import share from '@/assets/images/icons/share.svg';
+import hourglass from '@/assets/images/icons/hourglass.svg';
+import { useRouter } from 'vue-router';
+import * as echarts from 'echarts/core';
+import useVeBal from '@/composables/useVeBAL';
+import { useI18n } from 'vue-i18n';
+import { useHistoricalLocksQuery } from '@/composables/queries/useHistoricalLocksQuery';
+import { useLockRankQuery } from '@/composables/queries/useLockRankQuery';
+import VeBALMarketingHeader from './VeBALMarketingHeader.vue';
+
+/**
+ * COMPOSABLES
+ */
+const { account, isWalletReady } = useWeb3();
+
+const { isLoadingLockPool, isLoadingLockInfo, lock } = useLock();
+const { fNum } = useNumbers();
+const router = useRouter();
+
+const { veBalBalance } = useVeBal();
+const { t } = useI18n();
+const { isLoading, data: userHistoricalLocks } =
+  useHistoricalLocksQuery(account);
+const { isLoading: isLoadingLockBoard, data: userRankData } =
+  useLockRankQuery(account);
+
+/**
+ * COMPUTED
+ */
+const showVebalInfo = computed(() => {
+  const snapshots = userHistoricalLocks.value?.lockSnapshots;
+  return (
+    (isWalletReady.value && snapshots && snapshots.length > 0) ||
+    isLoadingData.value
+  );
+});
+
+const isLoadingData = computed(
+  () =>
+    isLoadingLockBoard.value ||
+    isLoading.value ||
+    isLoadingLockInfo.value ||
+    isLoadingLockPool.value
+);
+
+const lockedUntil = computed(() => {
+  if (lock.value?.hasExistingLock && !lock.value.isExpired) {
+    return format(lock.value.lockedEndDate, PRETTY_DATE_FORMAT);
+  }
+
+  return '—';
+});
+
+const chartValues = computed(() => {
+  if (!userHistoricalLocks.value) return [];
+
+  const { lockSnapshots } = userHistoricalLocks.value;
+  console.log('lockSnapshots', lockSnapshots);
+  const currentDate = (Date.now() / 1000).toFixed(0);
+
+  const chartV = lockSnapshots.reduce(
```

Comment:
Not a massive fan of big computed properties like this. Could you extract some logic into smaller functions? Could you also describe the transformation that's happening in this `reduce` so it's easier to understand for other devs?